### PR TITLE
Multiple files upload refactor

### PR DESCRIPTION
**Pull request by Alevsk**
- failed uploaded objects progress bar shows in red color
- fixed bug in where failed uploaded objects cannot be removed from
  listed objects in ObjectManager
- display delete button for failed upload objects
- display setErrorSnackMessage component after done uploading all
  objects with number of failed objects
- fixed race condition bug during multiple objects upload, now we are
  using Promise.allSettled to handle synchronization between uploads

Screenshot from 2022-03-23 14-40-03
Screenshot from 2022-03-23 14-41-37

Signed-off-by: Lenin Alevski <alevsk.8772@gmail.com>

### CODE REVIEW

**Comment by cniackz** [comment]
Tested locally, wanted to see the red color:

red-bar

Disconnected minio server while uploading big file to see the error and the new red color on the bar. If Permission Tests are passing then this PR looks good to me!

**Review by cniackz — APPROVED** [review]
If `Permission Tests` are passing then this PR looks good to me!

File: portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/ListObjects/ListObjects.tsx [hunk 1]
```diff
@@ -763,135 +763,135 @@ const ListObjects = ({
...
+          );
+          if (errors.length > 0) {
+            const totalFiles = uploadFilePromises.length;
+            const successUploadedFiles =
+              uploadFilePromises.length - errors.length;
+            const err: ErrorResponseHandler = {
+              errorMessage: "There were some errors during file upload",
+              detailedError: `Uploaded files ${successUploadedFiles}/${totalFiles}`,
+            };
+            console.log("upload results", results);
```
Comment (prakashsvmx) [inline portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/ListObjects/ListObjects.tsx:891]: We could remove this console log.

**Review by prakashsvmx — APPROVED** [review]
Tested. Changes look good to me.

**Review by adfost — APPROVED** [review]

**Pull request merged**